src/hooks/usePoolsHandler.tsx

```diff
@@ -1,21 +1,53 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useState, useCallback } from 'react'
 import { notifyError } from './useNotification'
-import { PoolsQueryDocument, PoolsQueryQuery, execute } from '../../.graphclient'
+import { ExecutionResult } from 'graphql'
+import { PoolsDocument, Pool, PoolsQuery, LatestPoolDataDocument, execute, PoolData } from '../../.graphclient'
 
 export const usePoolsHandler = () => {
-  const [poolsData, setPoolsData] = useState<PoolsQueryQuery>()
+  const [pools, setPools] = useState<ExecutionResult<PoolsQuery>>()
```

Comment:
We should use a more simple type that is not api specific to store in variables. ie, today we may use Graphql, tomorrow we may swtich to REST etc.

Reply:
what do you mean? We are using the types that are specific to the graphclient's `execute` function. I mean, isn't that what typescript should be? having specific types for a certain method? We aren't creating an agnostic API for people to use when querying information from the subgraph. If we want to change the method to something like REST, well we'll change it.

The `ExecutionResult` just makes sure that the data received from the subgraph is honest and matches the types that were generated, to prevent bad actors from malicious actions.

Reply:
This is not absolutely necessary, but it is just a matter of decoupling good practice.

Reply:
what other data objects would we be interacting with? also, this is not the final implementation for the queries. It will be improved